File: C/Coleman-Liau Index (Readability test)/readability.c
#include "cs50.h"
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int count_grade(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    int grade = count_grade(letters, words, sentences);

    if (grade >= 16) // checking the grade for 1-15
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text) // function for calculating the number of letters.
{
    int letters = 0;
    for (int i = 0; i <= strlen(text); i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letters++;
        }
        else
        {
            continue;
        }
    }
    return letters;
}

int count_words(string text) // function for calculating the number of words.
{
    int words = 0;
    for (int i = 0; i <= strlen(text); i++)
    {
        if (text[i] == 32)
        {
            words++;
        }
        else
        {
            continue;
        }
    }
    return words + 1; // the last word of the text doesn't have any space that's why.
}

int count_sentences(string text) // function for calculating the number of sentences.
{
    int sentences = 0;
    for (int i = 0; i <= strlen(text); i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        }
        else
        {
            continue;
        }
    }
    return sentences;
}

int count_grade(int letters, int words, int sentences) // function for calculating the grade.
{

    float l = ((float)letters / (float)words) * 100;
    float s = ((float)sentences / (float)words) * 100;

    int index = round(0.0588 * l) - (0.296 * s) - 15.8; // Coleman-Liau index formula
    // int index = round(value);

    return index;
}

File: Python/Readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = (letters / words) * 100
    S = (sentences / words) * 100

    grade = round(0.0588 * L - 0.296 * S - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")

    # print(l, ",", s, ",", w)


def count_letters(text):
    count = 0

    for i in text:
        if 32 <= ord(i) <= 47 or 58 <= ord(i) <= 64 or 91 <= ord(i) <= 96 or 123 <= ord(i) <= 126:
            continue
        else:
            count += 1

    return count


def count_words(text):
    count = 0

    for i in text:
        if ord(i) == 32:
            count += 1
        else:
            continue

    return count + 1


def count_sentences(text):
    count = 0

    for i in text:
        if ord(i) == 33 or ord(i) == 46 or ord(i) == 63:
            count += 1
        else:
            continue

    return count


if __name__ == "__main__":
    main()
